textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/source/ScanBasedSourceOperator.java

```diff
@@ -79,10 +85,29 @@ public ITuple getNextTuple() throws DataFlowException {
                 return null;
             }
             Document document = indexSearcher.doc(scoreDocs[cursor++].doc);
+            
             List<IField> fields = new ArrayList<IField>();
-            for (String fieldName : schema) {
-                //TODO Generalize it to all field types(int, date etc)
-                IField field = new StringField(document.get(fieldName));
+            for (Attribute  attr : schema) {
+                FieldType fieldType = attr.getFieldType();
+                IField field = null;
+                String fieldValue = document.get(attr.getFieldName());
+                switch (fieldType) {

```

Comment:
wanna make a utility function for these case matching? it seems will be used in many places.